eth2_testgen/bls/tgen_bls.py

```diff
@@ -0,0 +1,195 @@
+            })
+
+    case03_private_to_public_key = []
+    pubkeys = []  #  Used in later cases
+    pubkeys_serial = []  #  Used in public key aggregation
+    for privkey in PRIVKEYS:
+        pubkey_serial = '0x' + bls.privtopub(privkey).hex()
+        case03_private_to_public_key.append({
+            'input': int_to_hex(privkey),
+            'output': pubkey_serial
+        })
+        pubkeys_serial.append(pubkey_serial)
+
+    case04_sign_messages = []
+    sigs = []  # used in verify
+    for privkey in PRIVKEYS:
+        for message in MESSAGES:
+            for domain in DOMAINS:
+                sig = bls.sign(message, privkey, domain)
+                case04_sign_messages.append({
+                    'input': {
+                        'privkey': int_to_hex(privkey),
+                        'message': '0x' + msg.hex(),
+                        'domain': int_to_hex(domain)
+                    },
+                    'output': '0x' + sig.hex()
+                })
+                sigs.append(sig)
+
+    # TODO: case05_verify_messages: Verify messages signed in case04
+    # It takes too long, empty for now
+
+    case06_aggregate_sigs = []
+    for domain in DOMAINS:
+        for message in MESSAGES:
+            sigs = []
+            for privkey in PRIVKEYS:
+                sig = bls.sign(message, privkey, domain)
+                sigs.append(sig)
+            case06_aggregate_sigs.append({
+                'input': ['0x' + sig.hex() for sig in sigs],
+                'output': '0x' + bls.aggregate_signatures(sigs).hex(),
+            })
+
+    case07_aggregate_pubkeys = {
+        'input': pubkeys_serial,
+        'output': '0x' + bls.aggregate_pubkeys(pubkeys).hex(),
+    }
+
+    # TODO
+    # Aggregate verify
+
+    # TODO
+    # Proof-of-possession
+
+    with open(sys.argv[1], 'w') as outfile:
+        # Dump at top level
+        yaml.dump(metadata, outfile, default_flow_style=False)
+        # default_flow_style will unravel "ValidatorRecord" and "committee" line,
+        # exploding file size
+        yaml.dump(
+            {'case01_message_hash_G2_uncompressed': case01_message_hash_G2_uncompressed},
+            outfile,
+        )
+        yaml.dump(
+            {'case02_message_hash_G2_compressed': case02_message_hash_G2_compressed},
+            outfile,
+        )
+        yaml.dump(
+            {'case03_private_to_public_key': case03_private_to_public_key},
+            outfile,
+        )
+        yaml.dump(
+            {'case04_sign_messages': case04_sign_messages},
+            outfile,
+        )
+
+        # Too time consuming to generate
+        # yaml.dump({'case05_verify_messages': case05_verify_messages}, outfile)
+        yaml.dump(
+            {'case06_aggregate_sigs': case06_aggregate_sigs},
+            outfile,
+        )
+        yaml.dump(
+            {'case07_aggregate_pubkeys': case07_aggregate_pubkeys},
+            outfile,
+        )
```

Comment:
The dump part seems like a lot of extra lines

Reply:
Case 1~3 are due to linting and the [one-thing-per-line style](https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/style-guide.md#one-thing-per-line). I'll make the rest of them one-liner.